backend/src/controllers/tasks.controller.ts

```diff
@@ -0,0 +1,80 @@
+import { Request, Response } from 'express';
+import { Project, Task } from '../models/project';
+
+async function fetchTaskDetails(req: Request, res: Response) {
+  try {
+    const taskDetail = await Task.find();
+    if (!taskDetail) throw new Error('Not found a Task');
+    res.status(200).send(taskDetail);
+  } catch (error) {
+    res.status(400).send((error as Error).message);
+  }
+}
+async function fetchTaskDetailsByIdProject(req: Request, res: Response) {
+  try {
+    const taskDetails = await Task.find({ idProject: req.params.idProject });
+    if (!taskDetails) throw new Error('Not found a Project');
```

Comment:
please achange to this code
const {idProject} = req.params
in the top of the function